File: oneDconvelution.py
import time
def main():
    start = time.perf_counter()
    signal = [0] * 1024
    for i in range(1024):
        signal[i] = i + 1

    kernel = [1, 1, 1]
    result = [0] * (len(signal) - len(kernel) + 1)
    for i in range(len(result)):
        sum = 0 
        for j in range(len(kernel)):
            sum+=signal[i + j] * kernel[j]
        result[i] = (sum)
    end = time.perf_counter() - start
    print(end)

    # i = len(result) - 4
    # print(f"{result[0:3]}")
    # print(f"{result[i:i+4]}")

if __name__ == "__main__":
    main()
    #0.0035114089999979115 on rasberi
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.0038747840000041833
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.002382355000008829
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.0025603719999480745
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.0043800739999824145
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.002398541999923509
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.00239507899993896
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.0023719509999864385
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.002491301999953066
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.0024093039999115717
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.002555208999979186
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.0023519349999787664
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.0024790079999092995
# Pi@raspberrypi:~/Desktop/MetaMotionRLPython $ python3 oneDconvelution.py 
# 0.003934629000013956

File: oneDconvelution.cpp
#include <iostream>
#include <vector>
#include <chrono>

int main() {
    auto t0 = std::chrono::high_resolution_clock::now();
    std::vector<int> signal(1024);
    for (int i = 0; i < 1024; i++) {
        signal[i] = i + 1;
    }

    std::vector<int> kernel = {1, 1, 1};

    std::vector<int> result(signal.size() - kernel.size() + 1);

    for (size_t i = 0; i < result.size(); i++) {
        int sum = 0;
        for (size_t j = 0; j < kernel.size(); j++) {
            sum += signal[i + j] * kernel[j];
        }
        result[i] = sum;
    }
    auto t1 = std::chrono::high_resolution_clock::now();

    auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(t1 - t0);

    std::cout << "Elapsed time: "
              << elapsed.count() << " ms\n";

    return 0;
}